angular/projects/spark-angular/src/lib/components/sprk-autocomplete/sprk-autocomplete-result/sprk-autocomplete-result.directive.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { Component } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { SprkAutocompleteResultDirective } from './sprk-autocomplete-result.directive';
+import { By } from '@angular/platform-browser';
+
+@Component({
+  selector: 'sprk-test',
+  template: `
+    <ul>
+      <li
+        sprkAutocompleteResult
+        analyticsString="result1"
+        idString="resultId1"
+        additionalClasses="testClass"
+      ></li>
+      <li sprkAutocompleteResult></li>
+    </ul>
+  `,
+})
+class TestComponent {}
+
+describe('Spark Autocomplete Result Directive', () => {
+  let component: TestComponent;
+  let fixture: ComponentFixture<TestComponent>;
+  let element: HTMLElement;
+  let directiveElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [SprkAutocompleteResultDirective, TestComponent],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(TestComponent);
+    component = fixture.componentInstance;
+    element = fixture.nativeElement.querySelector('li');
+
+    directiveElement = fixture.debugElement
+      .query(By.directive(SprkAutocompleteResultDirective))
+      .injector.get(
+        SprkAutocompleteResultDirective,
+      ) as SprkAutocompleteResultDirective;
+
+    fixture.detectChanges();
+  }));
+
+  it('should create itself', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should add the correct base class', () => {
+    expect(element.classList.contains('sprk-c-Autocomplete__result')).toEqual(
+      true,
+    );
+  });
+
+  it('should add classes if additionalClasses has a value', () => {
+    expect(element.classList.contains('testClass')).toEqual(true);
```

Comment:
Lets do a stricter check here by not checking if it contains the class but by checking all of the classes on the element to make sure they are only what we expect.